modules/adding-bare-metal-host-to-cluster-using-yaml.adoc

```diff
@@ -0,0 +1,45 @@
+// Module included in the following assemblies:
+//
+// scalability_and_performance/managing-bare-metal-hosts.adoc
+
+[id="adding-bare-metal-host-to-cluster-using-yaml_{context}"]
+= Adding a bare metal host to the cluster using YAML in the web console
+
+You can add bare metal hosts to the cluster in the web console using a YAML file that describes the bare metal host.
+
+.Prerequisites
+
+* Install a {op-system} compute machine on bare metal infrastructure for use in the cluster.
+* Log in as a user with `cluster-admin` privileges.
+* Create a `Secret` CR for the bare metal host.
+
+.Procedure
+
+. In the web console, navigate to *Compute* -> *Bare Metal Hosts*.
+. Select *Add Host* -> *New from YAML*.
+. Copy and paste the below YAML, modifying the relevant fields with the details of your host:
++
+[source,yaml]
+----
+apiVersion: metal3.io/v1alpha1
+kind: BareMetalHost
+metadata:
+  name: <bare_metal_host_name>
+spec:
+  online: true
+  bmc:
+    address: <bmc_address>
+    credentialsName: <secret_credentials_name>
+    disableCertificateVerification: True
+    username: <secret_username>
+    password: <secret_password>
```

Comment:
These username/password fields don't actually exist in the BareMetalHost type. Is the UI doing something sneaky here to create the Secret in the background?

Reply:
@zaneb No, console is creating a secret unless user added the secret YAML.
 I tried to create a BareMetalHost with given example YAML on web console, finally username/password fields will be removed and no secret is created

Reply:
So should i leave remove the secret details altogether? eg just have following: 

```
apiVersion: metal3.io/v1alpha1
kind: BareMetalHost
metadata:
  name: <bare_metal_host_name>
spec:
  online: true
  bmc:
    address: <bmc_address>
    disableCertificateVerification: True
  bootMACAddress: <host_boot_mac_address>
  hardwareProfile: unknown
  ```

Reply:
You can create it with the credentialsName and create the Secret later, or create the Secret first, but the baremetal-operator won't be able to manage the host without the Secret existing and being referenced in the credentialsName.

Reply:
@yapei @zaneb made a change to support @zaneb's suggestion. LMK what you think.  https://deploy-preview-30656--osdocs.netlify.app/openshift-enterprise/latest/scalability_and_performance/managing-bare-metal-hosts.html#adding-bare-metal-host-to-cluster-using-yaml_managing-bare-metal-hosts